src/libs/ascent/utils/ascent_string_utils.cpp

```diff
@@ -39,39 +50,243 @@ void split_string(const std::string &s,
+std::string expand_generic_variable(const std::string& path_string, 
+                                    const std::regex& pattern, 
+                                    const T value)
+{
+    std::smatch match;
+    std::string result_string = path_string;
+
+    // Defining a valid int format to be any number of digits followed by a d,i, or u character
+    //
+    // The different supported integer number types are:
+    //    d: signed decimal integer
+    //    i: digned decimal integer
+    //    u: unsigned decimal integer
+    //
+    // e.g. 003d or 34i or 1d or 4u
+    std::regex valid_int_format(R"(^\d*[diu]$)");
+
+    // Defining a valid float format to be a real number followed by a f,e,g,F,E, or G character
+    // Real numbers are any number of digits potentially followed by a decimal point and at least one digit
+    //
+    // The different supported floating point number types are:
+    //    f: decimal floating point
+    //    F: decimal floating point
+    //    e: scientific notation using e to indicate the exponent
+    //    E: same as e except uses E to indicate the exponent
+    //    g: Uses the shortest notation (either e or f)
+    //    G: Uses the shortest notation (either E or F)
+    //
+    // e.g. 3.14f or 2F or 023g or 10.4e
+    std::regex valid_float_format(R"(^\d*(\.\d+)?[fFeEgG]$)");
+
+    while (std::regex_search(result_string, match, pattern))
     {
-      counter = s_file_family_map[name] + 1;
-      s_file_family_map[name] = counter;
+        std::string format_spec = match[1].str();
+
+        if (std::regex_match(format_spec, valid_int_format))
+        {
+          char formatted_number[50];
+          std::string full_format = "%" + format_spec;
+          snprintf(formatted_number, sizeof(formatted_number), full_format.c_str(), static_cast<int>(value));
+          result_string.replace(match.position(0), match.length(0), formatted_number);
+        }
+        else if (std::regex_match(format_spec, valid_float_format))
+        {
+          char formatted_number[50];
+          std::string full_format = "%" + format_spec;
+          snprintf(formatted_number, sizeof(formatted_number), full_format.c_str(), static_cast<float>(value));
+          result_string.replace(match.position(0), match.length(0), formatted_number);
+        }
+        else if (format_spec.size() == 0) {
+          ASCENT_WARN("No format specifications given. Inserting value without formatting.");
+          result_string.replace(match.position(0), match.length(0), std::to_string(value));
+        }
+        else
+        {
+          ASCENT_WARN("Invalid format specifier: '" 
+                        << format_spec 
+                        << "'. Inserting value without formatting.");
+          result_string.replace(match.position(0), match.length(0), std::to_string(value));
+        }
     }
+
+    return result_string;
+}
+
+int check_directory_for_family_value(const std::string& path_string, int family_value) {
```

Comment:
we should pass `int mpi_comm_id` id here. 

We can use a strategy like this to handle mpi vs non mpi:

https://github.com/Alpine-DAV/ascent/blob/7cb169c0ddf1f5217458cde2d6179ad9a67e9223/src/libs/ascent/utils/ascent_actions_utils.cpp#L368